<!doctype html>
<html>
<head>
	<meta charset="UTF-8">
	<title>NOX_CP12</title>
</head>
<body>
<h1>Searching for NOX homologues in “CP12”</h1>

<p>Here is a summary of the results from homology searches of NOX-like sequences against the “CP12” sequence database. BLAST searches were performed using protein sequences as queries against both the DNA (tblastn) and protein database (blastp). Additional searches were performed against the UniProt/SwissProt database to gain more information about target proteins.</p>

<table>
<colgroup>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
</colgroup>

<thead>
<tr>
	<th style="text-align:left;">Target locus</th>
	<th style="text-align:left;">Target annotation</th>
	<th style="text-align:left;">Length</th>
	<th style="text-align:left;">Query sequence(s)</th>
	<th style="text-align:left;">Query organism(s)</th>
	<th style="text-align:left;">Score (bits)</th>
	<th style="text-align:left;">Target prediction</th>
</tr>
</thead>

<tbody>
<tr>
	<td style="text-align:left;">09080</td>
	<td style="text-align:left;">hypothetical protein</td>
	<td style="text-align:left;">403</td>
	<td style="text-align:left;">Q9NRD9 (Dual oxidase 1), Q54F44 (Superoxide-generating NADPH oxidase)</td>
	<td style="text-align:left;"><em>Homo sapiens</em>, <em>Dictyostelium discoideum</em></td>
	<td style="text-align:left;">36.2, 30.4</td>
	<td style="text-align:left;">Calmodulin-like</td>
</tr>
<tr>
	<td style="text-align:left;">14850</td>
	<td style="text-align:left;">similar to aluminium resistance protein</td>
	<td style="text-align:left;">411</td>
	<td style="text-align:left;">Chro_4270, prokaryotic NADPH oxidase (pNOX)</td>
	<td style="text-align:left;"><em>Chroococcidiopsis thermalis</em></td>
	<td style="text-align:left;">33.9</td>
	<td style="text-align:left;"><mark>Ferredoxin-like Aluminium resistance protein ?</mark></td>
</tr>
<tr>
	<td style="text-align:left;">05450</td>
	<td style="text-align:left;">oxidoreductase FAD/NAD(P)-binding domain protein</td>
	<td style="text-align:left;">405</td>
	<td style="text-align:left;">Q82R51 (Oxidoreductase), 8QQ7 (FAD-binding FR-type domain-containing protein)</td>
	<td style="text-align:left;"><em>Streptomyces avermitilis</em>, <em>Streptococcus pneumoniae</em></td>
	<td style="text-align:left;">32.7, 30.1</td>
	<td style="text-align:left;">Ferredoxin-NADPH reductase (FNR)</td>
</tr>
<tr>
	<td style="text-align:left;">34440</td>
	<td style="text-align:left;">hypothetical protein</td>
	<td style="text-align:left;">648</td>
	<td style="text-align:left;">Q9NRD9, Dual oxidase 1</td>
	<td style="text-align:left;"><em>Homo sapiens</em></td>
	<td style="text-align:left;">32.7</td>
	<td style="text-align:left;">Membrane-bound protease</td>
</tr>
<tr>
	<td style="text-align:left;">41500</td>
	<td style="text-align:left;">hypothetical protein</td>
	<td style="text-align:left;">809</td>
	<td style="text-align:left;">8QQ7, FAD-binding FR-type domain-containing protein</td>
	<td style="text-align:left;"><em>Streptococcus pneumoniae</em></td>
	<td style="text-align:left;">31.2</td>
	<td style="text-align:left;"><mark>NOX-like?</mark></td>
</tr>
<tr>
	<td style="text-align:left;">20480</td>
	<td style="text-align:left;">hypothetical protein</td>
	<td style="text-align:left;">988</td>
	<td style="text-align:left;">G2WM87, Ferric chelate reductase (NADPH)</td>
	<td style="text-align:left;"><em>Saccharomyces cerevisiae</em></td>
	<td style="text-align:left;">30.8</td>
	<td style="text-align:left;">Membrane-bound unknown</td>
</tr>
<tr>
	<td style="text-align:left;">23250</td>
	<td style="text-align:left;">hypothetical protein</td>
	<td style="text-align:left;">283</td>
	<td style="text-align:left;">K0S9N3, EF-hand domain-containing protein</td>
	<td style="text-align:left;"><em>Thalassiosira oceanica</em></td>
	<td style="text-align:left;">30.4</td>
	<td style="text-align:left;">MFS transporter with extra domain</td>
</tr>
</tbody>
</table>

<h2>Query: DUOX from <em>Homo sapiens</em> - blastp</h2>

<p><code>‌blastp -query DUOX1_HUMAN.fasta  -db comp_0_CP12_prot_db -out results.txt</code></p>

<ul>
<li><p><strong>Query sequence:</strong></p>

<blockquote>
<p>DUOX1_HUMAN.fasta </p>
</blockquote></li>
</ul>

<pre><code>&gt;sp|Q9NRD9|DUOX1_HUMAN Dual oxidase 1 OS=Homo sapiens OX=9606
 GN=DUOX1 PE=1 SV=1
MGFCLALAWTLLVGAWTPLGAQNPISWEVQRFDGWYNNLMEHRWGSKGSRLQRLVPASYA
DGVYQPLGEPHLPNPRDLSNTISRGPAGLASLRNRTVLGVFFGYHVLSDLVSVETPGCPA
EFLNIRIPPGDPMFDPDQRGDVVLPFQRSRWDPETGRSPSNPRDPANQVTGWLDGSAIYG
SSHSWSDALRSFSRGQLASGPDPAFPRDSQNPLLMWAAPDPATGQNGPRGLYAFGAERGN
REPFLQALGLLWFRYHNLWAQRLARQHPDWEDEELFQHARKRVIATYQNIAVYEWLPSFL
QKTLPEYTGYRPFLDPSISSEFVAASEQFLSTMVPPGVYMRNASCHFQGVINRNSSVSRA
LRVCNSYWSREHPSLQSAEDVDALLLGMASQIAEREDHVLVEDVRDFWPGPLKFSRTDHL
ASCLQRGRDLGLPSYTKARAALGLSPITRWQDINPALSRSNDTVLEATAALYNQDLSWLE
LLPGGLLESHRDPGPLFSTIVLEQFVRLRDGDRYWFENTRNGLFSKKEIEEIRNTTLQDV
LVAVINIDPSALQPNVFVWHKGDPCPQPRQLSTEGLPACAPSVVRDYFEGSGFGFGVTIG
TLCCFPLVSLLSAWIVARLRMRNFKRLQGQDRQSIVSEKLVGGMEALEWQGHKEPCRPVL
VYLQPGQIRVVDGRLTVLRTIQLQPPQKVNFVLSSNRGRRTLLLKIPKEYDLVLLFNLEE
ERQALVENLRGALKESGLSIQEWELREQELMRAAVTREQRRHLLETFFRHLFSQVLDINQ
ADAGTLPLDSSQKVREALTCELSRAEFAESLGLKPQDMFVESMFSLADKDGNGYLSFREF
LDILVVFMKGSPEEKSRLMFRMYDFDGNGLISKDEFIRMLRSFIEISNNCLSKAQLAEVV
ESMFRESGFQDKEELTWEDFHFMLRDHNSELRFTQLCVKGVEVPEVIKDLCRRASYISQD
MICPSPRVSARCSRSDIETELTPQRLQCPMDTDPPQEIRRRFGKKVTSFQPLLFTEAHRE
KFQRSCLHQTVQQFKRFIENYRRHIGCVAVFYAIAGGLFLERAYYYAFAAHHTGITDTTR
VGIILSRGTAASISFMFSYILLTMCRNLITFLRETFLNRYVPFDAAVDFHRLIASTAIVL
TVLHSVGHVVNVYLFSISPLSVLSCLFPGLFHDDGSELPQKYYWWFFQTVPGLTGVVLLL
ILAIMYVFASHHFRRRSFRGFWLTHHLYILLYVLLIIHGSFALIQLPRFHIFFLVPAIIY
GGDKLVSLSRKKVEISVVKAELLPSGVTHLRFQRPQGFEYKSGQWVRIACLALGTTEYHP
FTLTSAPHEDTLSLHIRAAGPWTTRLREIYSAPTGDRCARYPKLYLDGPFGEGHQEWHKF
EVSVLVGGGIGVTPFASILKDLVFKSSVSCQVFCKKIYFIWVTRTQRQFEWLADIIREVE
ENDHQDLVSVHIYITQLAEKFDLRTTMLYICERHFQKVLNRSLFTGLRSITHFGRPPFEP
FFNSLQEVHPQVRKIGVFSCGPPGMTKNVEKACQLINRQDRTHFSHHYENF
</code></pre>

<ul>
<li><strong>Results:</strong></li>
</ul>

<table>
<colgroup>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
</colgroup>

<thead>
<tr>
	<th style="text-align:left;">Sequences producing significant alignments:</th>
	<th style="text-align:left;">Score (Bits)</th>
	<th style="text-align:left;">E-Value</th>
</tr>
</thead>

<tbody>
<tr>
	<td style="text-align:left;">LOCUS_09080 hypothetical protein</td>
	<td style="text-align:left;">36.2</td>
	<td style="text-align:left;">0.016</td>
</tr>
<tr>
	<td style="text-align:left;">LOCUS_34440 hypothetical protein</td>
	<td style="text-align:left;">32.7</td>
	<td style="text-align:left;">0.26</td>
</tr>
</tbody>
</table>

<h3>&gt; LOCUS_09080 hypothetical protein, Length = 403</h3>

<blockquote>
<p>Score = 36.2 bits (82), Expect = 0.016, Method: Compositional matrix adjust. Identities = 28/94 (30%), Positives = 44/94 (47%), Gaps = 15/94 (16%)</p>
</blockquote>

<pre><code>Query  799  TCELSRAEFA---ESLGLKPQDMFVESMFSLADKDGNGYLSFREF------LDILVVFMK  849
            +  +S  EFA   +S  +  QD  +   F L D DG+GY+S  E       LD LV    
Target  311  SSSISFEEFATLLQSNLVDDQDERLVERFKLFDADGSGYISLEELQVCIRDLDALVT---  367

Query  850  GSPEEKSRLMFRMYDFDGNGLISKDEFIRMLRSF  883
               + +   M +  D DG+G I+ +EF  + ++ 
Target  368  ---KAEVEAMLKGIDVDGDGQINYEEFYHLFQNL  398
</code></pre>

<blockquote>
<p>Score = 35.0 bits (79), Expect = 0.042, Method: Compositional matrix adjust. Identities = 23/105 (22%), Positives = 52/105 (50%), Gaps = 8/105 (8%)</p>
</blockquote>

<pre><code>Query  822  SMFSLADKDGNGYLSFREFLDILVVFMKGSPEEKSRLMFRMYDFDGNGLISKDEFIRMLR  881
            +M  +   + +  +SF EF  +L   +    +E+    F+++D DG+G IS +E    +R
Target  301  AMDLITGSENSSSISFEEFATLLQSNLVDDQDERLVERFKLFDADGSGYISLEELQVCIR  360

Query  882  SFIEISNNCLSKAQLAEVVESMFRESGFQDKEELTWEDFHFMLRD  926
                  +  ++KA+    VE+M +        ++ +E+F+ + ++
Target  361  DL----DALVTKAE----VEAMLKGIDVDGDGQINYEEFYHLFQN  397

</code></pre>

<ul>
<li><strong>Full length target sequence:</strong></li>
</ul>

<pre><code>&gt;LOCUS_09080 hypothetical protein
MLSNTNGSVQQELIRYMFRRDANGQYDAMACLQLWFGKSHDTDNEIRHRFGALVEQALAGELDHWMETP
HGCVALMILLDQFPRNLYRHTVQMYSGDQKAMQIVAQKHDWQAELTPEEYLFVPCLILTHQENLEAQKQ
CVNYYEQIEPILQPTFHIFRTIFEEHLKIINLCGVFPHRDHYYGRETSATGKMLMEDSTLRFDLPLVAE
NGSVRFGTDSTRLWKATAHAFDVLDRLDSLATQHQANDGKPVTIDWMTDEQIAFCKDTFRKFDKDGSGH
LSAAELRSVLQSVGRPYSAERIQAAMDLITGSENSSSISFEEFATLLQSNLVDDQDERLVERFKLFDAD
GSGYISLEELQVCIRDLDALVTKAEVEAMLKGIDVDGDGQINYEEFYHLFQNLKAQRA
</code></pre>

<ul>
<li><strong>Target homologues:</strong><br/>
<code>‌blastp -query LOCUS_09080.fasta  -db swissprot -remote</code></li>
</ul>

<table>
<colgroup>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
</colgroup>

<thead>
<tr>
	<th style="text-align:left;">Target accession</th>
	<th style="text-align:left;">Target sequence</th>
	<th style="text-align:left;">Organism</th>
	<th style="text-align:left;">Score (Bits)</th>
	<th style="text-align:right;">E-Value</th>
</tr>
</thead>

<tbody>
<tr>
	<td style="text-align:left;"><a href="https://www.ebi.ac.uk/interpro/protein/UniProt/Q7MF74/">P24044</a></td>
	<td style="text-align:left;">Calmodulin</td>
	<td style="text-align:left;"><em>‌ Plasmodium falciparum</em></td>
	<td style="text-align:left;">113</td>
	<td style="text-align:right;">5e-29</td>
</tr>
<tr>
	<td style="text-align:left;"><a href="https://www.uniprot.org/uniprotkb/A8I1Q0/entry">A8I1Q0</a></td>
	<td style="text-align:left;">Calmodulin</td>
	<td style="text-align:left;"><em>Heterocapsa triquetra</em></td>
	<td style="text-align:left;">44.7</td>
	<td style="text-align:right;">1e-28</td>
</tr>
<tr>
	<td style="text-align:left;"><a href="https://www.uniprot.org/uniprotkb/A3E3H0/entry">A3E3H0</a></td>
	<td style="text-align:left;">Calmodulin</td>
	<td style="text-align:left;"><em>Pfiesteria
piscicida</em></td>
	<td style="text-align:left;">111</td>
	<td style="text-align:right;">3e-28</td>
</tr>
<tr>
	<td style="text-align:left;"><a href="https://www.uniprot.org/uniprotkb/O02367/entry">O02367</a></td>
	<td style="text-align:left;">Calmodulin</td>
	<td style="text-align:left;"><em>Ciona intestinalis</em></td>
	<td style="text-align:left;">110</td>
	<td style="text-align:right;">7e-28</td>
</tr>
<tr>
	<td style="text-align:left;"><a href="https://www.uniprot.org/uniprotkb/P02594/entry">P02594</a></td>
	<td style="text-align:left;">Calmodulin</td>
	<td style="text-align:left;"><em>Electrophorus
electicus</em></td>
	<td style="text-align:left;">109</td>
	<td style="text-align:right;">2e-27</td>
</tr>
</tbody>
</table>

<p>Calmodulin acts as part of a calcium signal transduction pathway by mediating the control of a large number of enzymes, ion channels, aquaporins and other proteins through calcium-binding. Calcium-binding is required for the activation of calmodulin. Among the enzymes to be stimulated by the calmodulin-calcium complex are a number of protein kinases, such as myosin light-chain kinases and calmodulin-dependent protein kinase type II (CaMK2), and phosphatases. Generally, calmodulins are thought to be absent in prokaryotes.</p>

<figure>
<img src="LOCUS_09080.png" alt="" />
</figure>

<h3>&gt;LOCUS_34440 hypothetical protein, length=648</h3>

<blockquote>
<p>Score = 32.7 bits (73), Expect = 0.26, Method: Compositional matrix adjust. Identities = 31/105 (30%), Positives = 44/105 (42%), Gaps = 14/105 (13%) </p>
</blockquote>

<pre><code>Query  784  GTLPLDSSQKVREALTCELS-RAEFAESLGLKPQDMFVESMFSLADKDGN---GYLSFRE 839
            G     ++Q +RE LT EL  R  + +     P D  +    + A+  GN   G+L  R 
Target  394  GHFSYGTAQVIREPLTGELRWRLTYHQVYAHNP-DGIISGKVNWAEYMGNLERGWLGSRP 452

Query  840  FLDILVVFMKGSPEEKSRLMFRMYDFDGNGLISKDEFIRMLRSFI  884
              D L+ F   SP  +       YDF GN L    EF   L+  +
Target  453  VADALIRF---SPVTED------YDFGGNKLSVMGEFTTQLQVMM  488
</code></pre>

<ul>
<li><strong>Full length target sequence:</strong></li>
</ul>

<pre><code>&gt;LOCUS_34440 hypothetical protein
MPRMPEKKWWLPIYFLVGLIIVIALGIIISTWNSSNAPPTERTSNYSLFRKQNQLTFENYYTGNYPPNL
YQPVAPWTGRLILPDVSIRNKQVNNQTNNQNPINSVTNNQTSKSLGNQSVFFEVHHADGQHQNLRGRIV
NLQWNNQTLVNDYRRRTTRDISFTTKTWQSIKSGNIHPDRLHNWQAVDPLESLTGARPEDDLMVSIAPV
RVVESPQGDITLVIASEPEQVVGSFYGLVQIIAREKNDHFLVQHFNQVSHQFDGSEEVMIMPQLVVDRE
GVWRSTNQDIEKSPFNTTGWYVYGAKNSQGLFTVQALEPREILQVPPQIIQATPKEKKNYLEQQLWQNL
QKNTGSSVLLTNPTPAPWQEGDRAIVMHIFGGIGGKKAEPKALGMFVPGHFSYGTAQVIREPLTGELR
WRLTYHQVYAHNPDGIISGKVNWAEYMGNLERGWLGSRPVADALIRFSPVTEDYDFGGNKLSVMGEFTT
QLQVMMARYRTGSGTGAALVTPATSCVQDSNQALYITIQKITNQVAQNPQIQTWLQANPNHPQTQRFNQ
LITLGKALEQELVPLGIIRADWQEQVTEGTGTKRDSSVLGDFFSGITTWRTLLPRPAFDRLAQIFLDQG
ADLWFLRTNQVGGNDPDILPLAPTAPFS 
</code></pre>

<ul>
<li><strong>Target homologues:</strong></li>
</ul>

<p><code>‌‌blastp -query LOCUS_34440.fasta  -db swissprot -remote</code><br/>
No hits found in manually curated protein databse (SwissProt).
Expanded search on non reduntant genome database (nr) using protein sequence as query against translated genome database (tblastn):<br/>
<code>
Program: TBLASTN 
Query: LOCUS_34440 hypothetical protein ID: lcl|Query_8219019(amino 
acid) Length: 648  Database: nt Nucleotide collection (nt)
</code></p>

<table>
<colgroup>
<col style="text-align:center;"/>
<col style="text-align:center;"/>
<col style="text-align:center;"/>
<col style="text-align:center;"/>
<col style="text-align:center;"/>
<col style="text-align:center;"/>
</colgroup>

<thead>
<tr>
	<th style="text-align:center;"><strong>Organism</strong></th>
	<th style="text-align:center;"><strong>Total Score</strong></th>
	<th style="text-align:center;"><strong>Query Coverage</strong></th>
	<th style="text-align:center;"><strong>E-value</strong></th>
	<th style="text-align:center;">**Identify(%) **</th>
	<th style="text-align:center;"><strong>Accession</strong></th>
</tr>
</thead>

<tbody>
<tr>
	<td style="text-align:center;"><em>Moorena producens</em> PAL-8-15-08-1</td>
	<td style="text-align:center;">570</td>
	<td style="text-align:center;">93%</td>
	<td style="text-align:center;">3E-176</td>
	<td style="text-align:center;">47.33</td>
	<td style="text-align:center;">CP017599.1</td>
</tr>
<tr>
	<td style="text-align:center;"><em>Moorena producens</em> JHB</td>
	<td style="text-align:center;">567</td>
	<td style="text-align:center;">93%</td>
	<td style="text-align:center;">2E-175</td>
	<td style="text-align:center;">47.10</td>
	<td style="text-align:center;">CP017708.2</td>
</tr>
<tr>
	<td style="text-align:center;"><em>Trichocoleus desertorum</em> NBK24</td>
	<td style="text-align:center;">561</td>
	<td style="text-align:center;">89%</td>
	<td style="text-align:center;">2E-173</td>
	<td style="text-align:center;">48.20</td>
	<td style="text-align:center;">CP116619.1</td>
</tr>
<tr>
	<td style="text-align:center;"><em>Nostoc sp.</em> C052</td>
	<td style="text-align:center;">555</td>
	<td style="text-align:center;">94%</td>
	<td style="text-align:center;">5E-171</td>
	<td style="text-align:center;">46.12</td>
	<td style="text-align:center;">CP040272.1</td>
</tr>
<tr>
	<td style="text-align:center;"><em>Nostocales cyanobacterium</em> HT-58-2</td>
	<td style="text-align:center;">1005</td>
	<td style="text-align:center;">93%</td>
	<td style="text-align:center;">9E-167</td>
	<td style="text-align:center;">46.49</td>
	<td style="text-align:center;">CP019636.1</td>
</tr>
<tr>
	<td style="text-align:center;"><em>Nostoc sp.</em> UHCC 0870</td>
	<td style="text-align:center;">540</td>
	<td style="text-align:center;">93%</td>
	<td style="text-align:center;">1E-165</td>
	<td style="text-align:center;">47.52</td>
	<td style="text-align:center;">CP091913.1</td>
</tr>
<tr>
	<td style="text-align:center;"><em>Nostoc punctiforme</em> PCC 73102</td>
	<td style="text-align:center;">531</td>
	<td style="text-align:center;">93%</td>
	<td style="text-align:center;">9E-163</td>
	<td style="text-align:center;">45.53</td>
	<td style="text-align:center;">CP001037.1</td>
</tr>
<tr>
	<td style="text-align:center;"><em>Microcoleus vaginatus</em> PCC 9802</td>
	<td style="text-align:center;">530</td>
	<td style="text-align:center;">89%</td>
	<td style="text-align:center;">2E-162</td>
	<td style="text-align:center;">46.31</td>
	<td style="text-align:center;">CP031740.1</td>
</tr>
<tr>
	<td style="text-align:center;"><em>Trichormus variabilis</em> NIES-23</td>
	<td style="text-align:center;">530</td>
	<td style="text-align:center;">92%</td>
	<td style="text-align:center;">3E-162</td>
	<td style="text-align:center;">45.36</td>
	<td style="text-align:center;">AP018216.1</td>
</tr>
<tr>
	<td style="text-align:center;"><em>Nostoc sp.</em> PCC 7120 = FACHB-418</td>
	<td style="text-align:center;">530</td>
	<td style="text-align:center;">92%</td>
	<td style="text-align:center;">3E-162</td>
	<td style="text-align:center;">45.36</td>
	<td style="text-align:center;">BA000019.2</td>
</tr>
<tr>
	<td style="text-align:center;"><em>Calothrix sp.</em> PCC 7507</td>
	<td style="text-align:center;">528</td>
	<td style="text-align:center;">93%</td>
	<td style="text-align:center;">1E-161</td>
	<td style="text-align:center;">45.45</td>
	<td style="text-align:center;">CP003943.1</td>
</tr>
<tr>
	<td style="text-align:center;"><em>Nostoc sp.</em> PCC 7524</td>
	<td style="text-align:center;">527</td>
	<td style="text-align:center;">94%</td>
	<td style="text-align:center;">2E-161</td>
	<td style="text-align:center;">46.98</td>
	<td style="text-align:center;">CP003552.1</td>
</tr>
<tr>
	<td style="text-align:center;"><em>Trichormus variabilis</em> ATCC 29413</td>
	<td style="text-align:center;">526</td>
	<td style="text-align:center;">92%</td>
	<td style="text-align:center;">5E-161</td>
	<td style="text-align:center;">44.74</td>
	<td style="text-align:center;">CP000117.1</td>
</tr>
<tr>
	<td style="text-align:center;">…</td>
	<td style="text-align:center;">…</td>
	<td style="text-align:center;">…</td>
	<td style="text-align:center;">…</td>
	<td style="text-align:center;">…</td>
	<td style="text-align:center;">…</td>
</tr>
<tr>
	<td style="text-align:center;"><em>Synechocystis sp.</em> PCC 6803</td>
	<td style="text-align:center;">436</td>
	<td style="text-align:center;">89%</td>
	<td style="text-align:center;">1E-129</td>
	<td style="text-align:center;">42.16</td>
	<td style="text-align:center;">BA000022.2</td>
</tr>
</tbody>
</table>

<p>Homologue in <em>Synechocystis sp.</em> PCC 6803 is encoded in locus:<a href="https://www.genome.jp/entry/syn:sll1477"> sll1477</a> (UniProt: <a href="https://www.uniprot.org/uniprotkb/P74619/entry">P74619</a>) and in <em>Nostoc sp.</em> PCC 7120 is alr1712 (UniProt: <a href="https://www.uniprot.org/uniprotkb/Q8YWA3/entry">Q8YWA3</a>). Both are uncharacterised proteins annotated as “Predicted protease, Abi (CAAX) family “. Target contains transmembrane domains but no known Pfam domains.</p>

<figure>
<img src="LOCUS_34440.png" alt="" />
</figure>

<h2>Query: EF-hand domain-containing protein from <em>Thalassiosira oceanica</em> - blastp</h2>

<p><code>‌blastp -query NOX_Thala.fasta  -db comp_0_CP12_prot_db -out results.txt</code></p>

<ul>
<li><p><strong>Query sequence:</strong> </p>

<blockquote>
<p>NOX_Thala.fasta</p>
</blockquote></li>
</ul>

<pre><code>&gt;tr|K0S9N3|K0S9N3_THAOC EF-hand domain-containing protein 
OS=Thalassiosira oceanica OX=159749 GN=THAOC_24795 PE=4 SV=1
MTVTELCSAEDSGDYDNQRGLLARTYPLPPVPMRPRHLASQAEKGKHAEIEASFNFSIEN
QAQAQDAEIQQASIERARRLGPDGPAPHPPDPARVDVAASDSLRRSSDGSDRGEGDVEAP
PPTYRPSRKPRQQQPALEGRVCSVGDGGDDGEHDIESGRAASDGVVKSFRMGRSGSESGS
EHVRSQTDGRLALLEMSRMPTRRISNAAESTLSEESAETSGAASSITSTSSALDHTTRTI
KTASTSMRPKYNRRASNHLAASVRMDQLRILQTQLALMIEDAEVDDGSNDSSLENSLRRA
SDILRRSSDGSDRGECDVEAPPPTYRPSRKPRQQQPALEGSVCSVGDGGGGDGEHDIESG
RAASDGVIKSFRMGRSGSEHVRSQPVAAEVVDGRLAQLEMSRMPTRRISNAAESTLSEES
AETSGAASSITSTSSALDHTTRTIKTASTSGSNSLERPSRLRRRFKSSSPSRSPSNSRSK
SPLKKGKRNRSKTPAQRTFPIFGGRRKTSNAAREETLPSYEFLLQNEGGSKEEGSLLPNT
KWWHGVFFFSLVAFLACIVTLWAPYPIGARMLSAQVAEMPWSNGCQGIDTCICPRETICA
DDLLSMIFLTIARCSAWFDYPLYMLMFLSKANNLNHALQRTALRVFVNFSDQHKVHSLFG
VIVAIESTSHAFFHTLRWARRKNDIQLLWMSKTGVTGMIAMTFLPLIVLPMTVPFLKKRM
RFEWRKYLHYLSILWAVALMCHAPQRIFWMIGVPLFIYAADKLVEGMFKTHLIENCHFQR
LGETSCIVEFENPSGFGKQNSAYVYLMLPWLSRYQYHAFTVFPSSKPNHSTCCIYKTGDW
TGELLRNIAVPAHKPAFVVGPYLSPFSSPAMDSENLVAVASGIGVTPAFSLIRQYSLTSR
RLNLVWICRDPGLVEHWIQNINFGKNGYVLIYYTGKERKLIIPNSLPSNVLIFNGRPDLE
RTISGIIHAIASGEGLPEEMSKKVVTQTPAELRSKLLLEKALSLYSFDQLYEYTAKASNY
YNDTGEELVNMLNYQGVLSTMRHLLGGDAARVEKRIVENFERVDSDGDCLVDHSQFKEFF
DLMLMTDGNEEAESLENARSGIKKMSMCQDIFESGTGIAKDESKDEFGIMQYLRGDANGR
FQSKNWSMLYCGGSQPVLDSLETYKKKYGIKLSVEKFDW
</code></pre>

<p><strong>Results:</strong> </p>

<table>
<colgroup>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
</colgroup>

<thead>
<tr>
	<th style="text-align:left;">Sequences producing significant alignments:</th>
	<th style="text-align:left;">Score (Bits)</th>
	<th style="text-align:left;">E-Value</th>
</tr>
</thead>

<tbody>
<tr>
	<td style="text-align:left;">LOCUS_23250 hypothetical protein</td>
	<td style="text-align:left;">30.4</td>
	<td style="text-align:left;">0.73</td>
</tr>
</tbody>
</table>

<h3>&gt;LOCUS_23250 hypothetical protein, length=283</h3>

<blockquote>
<p>Score = 30.4 bits (67), Expect = 0.73, Method: Compositional matrix adjust. Identities = 25/113 (22%), Positives = 47/113 (42%), Gaps = 6/113 (5%)</p>
</blockquote>

<ul>
<li><strong>Alignment:</strong></li>
</ul>

<pre><code>Query  495  AQRTFPIFGGRRKTSNAAREETLP-SYEFLLQNEGGSKEEGSLLPNTKWW-HGVFFFSLV  552
            A R+ P+   R     A  +   P S +   Q   G +   +  P + WW   +FFF   
Target  120  ANRSIPVSTNRVADFTAQPQRVPPPSPQTKTQKSSGKRVNPTFAPPSPWWLRSIFFFKYS  179

Query  553  AFLACIVTLWAPYPIGARMLSAQVAEMPWSNGCQGIDTC-ICPRETICADDLL  604
            +  A +  + A       ++SAQ     W+   Q ++T     R+ + A++++
Target  180  SGFAVVGLIMAVLTTYGGIVSAQRQ---WTKEFQQLETLRQTERQLVAANEVI  229
</code></pre>

<ul>
<li><strong>Full length target sequence:</strong></li>
</ul>

<pre><code>&gt;LOCUS_23250 hypothetical protein
MSFAPRPLSVPAYQRRKRFGSIGSTFHANGANGANSANSANGAKKVAPNSSSRNPAANSPQVPPPHNF
QPPIPPTITPINKAKATRVNQANNRSTYPVNSYSSFHPINPINPINNGVNSANRSIPVSTNRVADFTA
QPQRVPPPSPQTKTQKSSGKRVNPTFAPPSPWWLRSIFFFKYSSGFAVVGLIMAVLTTYGGIVSAQRQ
WTKEFQQLETLRQTERQLVAANEVIKNQVAKQSVGNQELVKVDPNSRIYIPAPDPIVLATMKPPTPTP
SQSPKVIPLGY
</code></pre>

<ul>
<li><strong>Target homologues:</strong></li>
</ul>

<table>
<colgroup>
<col/>
<col/>
<col/>
<col/>
<col/>
<col/>
</colgroup>

<thead>
<tr>
	<th>Description</th>
	<th>Scientific Name</th>
	<th>Total Score</th>
	<th>Query Cover</th>
	<th>E value</th>
	<th>Identity (%)</th>
</tr>
</thead>

<tbody>
<tr>
	<td>hypothetical protein</td>
	<td>Coleofasciculaceae cyanobacterium SM2_1_6</td>
	<td>331</td>
	<td>85%</td>
	<td>8E-111</td>
	<td>72.87</td>
</tr>
<tr>
	<td>hypothetical protein</td>
	<td>Myxacorys almedinensis</td>
	<td>90.1</td>
	<td>60%</td>
	<td>1E-17</td>
	<td>36.36</td>
</tr>
<tr>
	<td>hypothetical protein</td>
	<td>Microcoleus sp. FACHB-SPT15</td>
	<td>84.7</td>
	<td>46%</td>
	<td>1E-15</td>
	<td>38.52</td>
</tr>
<tr>
	<td>hypothetical protein</td>
	<td>Lusitaniella coriacea</td>
	<td>84.3</td>
	<td>50%</td>
	<td>2E-15</td>
	<td>35.76</td>
</tr>
<tr>
	<td>hypothetical protein</td>
	<td>unclassified Microcoleus</td>
	<td>84.7</td>
	<td>41%</td>
	<td>2E-15</td>
	<td>39.34</td>
</tr>
<tr>
	<td>hypothetical protein</td>
	<td>Leptolyngbya sp. FACHB-711</td>
	<td>85.5</td>
	<td>47%</td>
	<td>2E-15</td>
	<td>38.41</td>
</tr>
<tr>
	<td>hypothetical protein</td>
	<td>Plectolyngbya sp. WJT66-NPBG17</td>
	<td>82.8</td>
	<td>41%</td>
	<td>3E-15</td>
	<td>45.38</td>
</tr>
<tr>
	<td>hypothetical protein</td>
	<td>Crinalium epipsammum</td>
	<td>83.6</td>
	<td>52%</td>
	<td>3E-15</td>
	<td>37.33</td>
</tr>
<tr>
	<td>hypothetical protein</td>
	<td>Cyanobacteria bacterium FACHB-502</td>
	<td>81.6</td>
	<td>42%</td>
	<td>6E-15</td>
	<td>40.00</td>
</tr>
<tr>
	<td>hypothetical protein</td>
	<td>unclassified Trichocoleus</td>
	<td>82.0</td>
	<td>47%</td>
	<td>9E-15</td>
	<td>35.25</td>
</tr>
</tbody>
</table>

<p>Most similar proteins to target are all cyanobacteria. There are no known characterised homologues and no known protein domains. Hard to say what this protein is.</p>

<figure>
<img src="LOCUS_23250%202.png" alt="" />
</figure>

<h2>Query: prokaryotic NOX (pNOX) from <em>Streptococcus pneumoniae</em></h2>

<p><code>‌blastp -query SpNOX_8QQ7.fasta  -db comp_0_CP12_prot_db -out results.txt</code></p>

<ul>
<li><p><strong>Query sequence:</strong> </p>

<blockquote>
<p>SpNOX_8QQ7.fasta </p>
</blockquote></li>
</ul>

<pre><code>&gt;8QQ7_1|Chain A|FAD-binding FR-type domain-containing protein|
Streptococcus pneumoniae (1313), Length=400
MEFSMKSVKGLLFIIASFILTLLTWMNTSPQFMIPGLALTSLSLTFILATRLPLLESWFHSLEKVYTV
HKFTAFLSIILLIFHNFSMGGLWGSRLAAQFGNLAIYIFASIILVAYLGKYIQYEAWRWIHRLVYLAY
ILGLFHIYMIMGNRLLTFNLLSFLVGSYALLGLLAGFYIIFLYQKISFPYLGKITHLKRLNHDTREIQ
IHLSRPFNYQSGQFAFLKIFQEGFESAPHPFSISGGHGQTLYFTVKTSGDHTKNIYDNLQAGSKVTLD
RAYGHMIIEEGRENQVWIAGGIGITPFISYIREHPILDKQVHFYYSFRGDENAVYLDLLRNYAQKNPN
FELHLIDSTKDGYLNFEQKEVPEHATVYMCGPISMMKALAKQIKKQNPKTELIYEGWKFK

</code></pre>

<p><strong>Results:</strong> </p>

<table>
<colgroup>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
</colgroup>

<thead>
<tr>
	<th style="text-align:left;">Sequences producing significant alignments:</th>
	<th style="text-align:left;">Score (Bits)</th>
	<th style="text-align:left;">E-Value</th>
</tr>
</thead>

<tbody>
<tr>
	<td style="text-align:left;">LOCUS_41500 hypothetical protein</td>
	<td style="text-align:left;">31.2</td>
	<td style="text-align:left;">0.14</td>
</tr>
<tr>
	<td style="text-align:left;">LOCUS_05450 oxidoreductase FAD/NAD(P)-binding domain protein</td>
	<td style="text-align:left;">30.8</td>
	<td style="text-align:left;">0.15</td>
</tr>
</tbody>
</table>

<h3>&gt;LOCUS_41500 hypothetical protein, length=809</h3>

<blockquote>
<p>Score = 31.2 bits (69), Expect = 0.14, Method: Compositional matrix adjust. Identities = 28/100 (28%), Positives = 51/100 (51%), Gaps = 16/100 (16%)</p>
</blockquote>

<pre><code>Query  27   NTSPQFMIPGLALTSLSL-TFILATRLPLLESWFHSLEK------VYTVHKFTAFLSII-  78
            +T+P F + G+ + ++ L +  +AT  PL++ WF   +K      +Y      + LS++ 
Target  96   DTNPVFWLLGILVITVGLPSLAIATTAPLMQRWFAIAQKNNDPYFLYAASNIGSLLSLVS  155

Query  79   --LLIFHNFSMGG---LWGSR---LAAQFGNLAIYIFASI  110
              +LI  NFS+     LW +    LA+  G  AI ++ S+
Target  156  YPILIETNFSLSQQSYLWAAGYGILASLTGICAISLWKSV  195
</code></pre>

<ul>
<li><strong>Full length protein sequence:</strong></li>
</ul>

<pre><code>&gt;LOCUS_41500 hypothetical protein  
MVIIYALTLFISAFLLFWVQLWVGKLLLPLLGGAPGVWNTCMVFFQALLLLGYSYAHVLSTRLKFRYQA
IVHTLFLLIPLSLLPLSLSPATSPPQDTNPVFWLLGILVITVGLPSLAIATTAPLMQRWFAIAQKNNDP
YFLYAASNIGSLLSLVSYPILIETNFSLSQQSYLWAAGYGILASLTGICAISLWKSVAPEQISLLADSS
ILDVNLVNENLGQPSKSQQFRWITLAFLPASLLLGTTTYITTDIAAVPLFWAIPLIVYLSTFILAFTNN
SLINQQFNYQKLLPLLPALLTTLVISFLLQLMRPIALLVPLHLVGLFIVGYICHGELASDRPKPNYLTL
FYFWISLGGVAGGIFNAIISPLLFTSVLEYPLVLIISFLLLPKLATVSNSGKTSELTPELISRLSTELN
SNESISKSISSFIPKPTTEANNSRFRFSIPISLGLILATLIVGFSPRFVWSDWLGIILSIMLWLGIYKT
FNLSKIRLLVGLLLLSLLGQFSLTTMGQVLVSDRSFFGIYRVVSQQQGQYYSLFHGTTLHGKQSRDPQR
RNEPLTYFSTTGPIGQVFNYFSRKDTLQSSGQSLEKFIDNNSQNSHIAVLGLGVGTLFSYARSPQEWTF
YEIDPVVVKLATDPNYFSFLSNDFSNSQEAIKKQIILGDGRLQLAKAKDDYYDLLVMDAFSSDAIPTHL
VTLEAMQLYLQKLRKKGIIAVNISNRFINLEPVLGKLAQELNLVSYSQVDRAITTQEKEQGKTASHWVI
MARNQESFGSLITDFSWQNMSTQSNSNFKVWTDDFSDIWQVLRLNERESS

</code></pre>

<ul>
<li><strong>InterProScan search results:</strong><br/>
The N-terminus contains 14 predicted transmembrane domains and a soluble C terminal methyltransferase domain / spermidine synthase.<br/>
No NAD(P)H / FAD binding domains predicted.</li>
</ul>

<figure>
<img src="LOCUS_41500.png" alt="" />
</figure>

<ul>
<li><strong>Target homologues:</strong><br/>
<code>blastp -query LOCUS_41500.fasta  -db uniprot-swiss</code></li>
</ul>

<table>
<colgroup>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
</colgroup>

<thead>
<tr>
	<th style="text-align:left;">Target accession</th>
	<th style="text-align:left;">Target sequence</th>
	<th style="text-align:left;">Organism</th>
	<th style="text-align:left;">Score (Bits)</th>
	<th style="text-align:right;">E-Value</th>
</tr>
</thead>

<tbody>
<tr>
	<td style="text-align:left;"><a href="https://www.ebi.ac.uk/interpro/protein/UniProt/Q7MF74/">Q7MF74</a></td>
	<td style="text-align:left;">Polyamine aminopropyltransferase</td>
	<td style="text-align:left;"><em>Vibrio vulnificus</em></td>
	<td style="text-align:left;">55.1</td>
	<td style="text-align:right;">1e-06</td>
</tr>
<tr>
	<td style="text-align:left;"><a href="https://www.uniprot.org/uniprotkb/Q5B467/entry">Q5B467</a></td>
	<td style="text-align:left;">N-terminal histidine N-methyltransferase</td>
	<td style="text-align:left;"><em>Emericella nidulans</em></td>
	<td style="text-align:left;">44.7</td>
	<td style="text-align:right;">0.003</td>
</tr>
<tr>
	<td style="text-align:left;"><a href="https://www.uniprot.org/uniprotkb/Q8G4L4/entry">Q8G4L4</a></td>
	<td style="text-align:left;">Probable polyamine aminopropyl transferase</td>
	<td style="text-align:left;"><em>Bifidobacterium longum</em></td>
	<td style="text-align:left;">42.4</td>
	<td style="text-align:right;">0.017</td>
</tr>
<tr>
	<td style="text-align:left;"><a href="https://www.uniprot.org/uniprotkb/A1TZ62/entry">A1TZ62</a></td>
	<td style="text-align:left;">Ion-translocating oxidoreductase complex</td>
	<td style="text-align:left;"><em>Marinobacter nauticus</em></td>
	<td style="text-align:left;">37.4</td>
	<td style="text-align:right;">0.49</td>
</tr>
</tbody>
</table>

<pre><code>&gt;sp|Q7MF74|SPEE_VIBVY Polyamine aminopropyltransferase OS=Vibrio 
vulnificus (strain YJ016) OX=196600 GN=speE PE=3 SV=1
Length=300

 Score = 55.1 bits (131),  Expect = 1e-06, Method: Compositional matrix adjust.
 Identities = 40/127 (31%), Positives = 62/127 (49%), Gaps = 19/127 (15%)

Query  593  NSQNSHIAVLGLGVGTLFS-----YARSPQEWTFYEIDPVVVKLATDPNYFSFLSNDFSN  647
            N +   I ++GLG G++ S     Y  S       EIDPVV ++A    YF F       
Target  88   NPKPERILIIGLGGGSVPSALAELYPES--HMDIVEIDPVVSQIAE--RYFYF-------  136

Query  648  SQEAIKKQIILGDGRLQLAKA--KDDYYDLLVMDAFSSDAIPTHLVTLEAMQLYLQKLRK  705
             Q +   ++  GD R+ + +A  K   YD +++DAF+ + IP HL+T E +    Q L  
Target  137  -QPSHNTRVHTGDARVYIKRAGLKGQKYDFILLDAFNGEYIPEHLMTREFLMETKQLLSS  195

Query  706  KGIIAVN  712
             G++  N
Target  196  SGVLVAN  202

</code></pre>

<ul>
<li><strong>Polyamine aminopropyltransferase</strong>: catalyses the irreversible transfer of a propylamine group from the amino donor S-adenosylmethioninamine (decarboxy-AdoMet) to putrescine (1,4-diaminobutane) to yield spermidine.</li>
<li><strong>Protein-histidine N-methyltransferase</strong>: specifically mediates 3-methylhistidine (tele-methylhistidine) methylation at 'His-1', which protects the side-chain from oxidative damage <a href="https://doi.org/10.1038/s41467-023-39875-7">(Batth et al., 2023)</a>. It is a multi-pass membrane protein (7 TM domains like NOXs) found in the endoplasmic reticulum membrane.</li>
<li><strong>Ion-translocating oxidoreductase complex:</strong> Part of a membrane-bound complex that couples electron transfer with translocation of ions across the membrane. This family of bacterial proteins includes a sodium-translocating NADH-ubiquinone oxidoreductase (i.e. a respiration linked sodium pump). The similarity of these proteins to NADH-ubiquinone oxidoreductases was previously noted by <a href="https://doi.org/10.1021/bi970014q">Kumagai et al (1997)</a>.</li>
</ul>

<h3>&gt;LOCUS_05450 oxidoreductase FAD/NAD(P)-binding domain protein, length=405</h3>

<blockquote>
<p>Score = 30.8 bits (68), Expect = 0.15, Method: Compositional matrix adjust. Identities = 19/62 (31%), Positives = 31/62 (50%), Gaps = 5/62 (8%)</p>
</blockquote>

<pre><code> Query  262  NLQAGSKVTLDRAYGHMII--EEGRENQVWIAGGIGITPFISYIREHPILDKQVHFYYSF  319
            NLQ G  + +    G  ++  E+   N + +A G GI PF +Y+     + K+ H  Y F
Target  229  NLQPGDDLKITGPVGKEMLLPEDPESNVIMMATGTGIAPFRAYLWR---MFKETHEDYKF  285

Query  320  RG  321
            +G
Target  286  KG  287
</code></pre>

<p>This is the same target found with <em>S. avermilitis</em>. Probably FNR.</p>

<h2>Query: NOX5 from <em>Cylindrospermum stagnale</em> PCC 7417</h2>

<p><code>blastp -query csNOX5_pdb5O0X.fasta -db comp_0_CP12_prot_db -out results.txt</code> </p>

<ul>
<li><p><strong>Query sequence:</strong> </p>

<blockquote>
<p>csNOX5_pdb5O0X.fasta</p>
</blockquote></li>
</ul>

<pre><code>&gt;5O0X_1|Chain A|Putative ferric reductase|Cylindrospermum stagnale
PCC 7417 (56107), Length=291
GSEPTFVVNASLLPSKVLGLQVQRPQSFNYQPGDYLFIKCPGISKFEWHPFTISSAPEMPDVLTLHIRAVGSWTG
KLYQLIREQREEWIRSGSSQSLPGVPVYIDGPYGTPSTHIFESKYAILICAGIGVTPFASILKSILHRNQQNPAK
MPLKKVHFYWLNREQKAFEWFVELLSKIEAEDTNNLFDLNLYLTGAQQKSDMKSSTLFVAMDLMHQETKVDLITG
LKSRTKTGRPDWEEIFKDVAKQHAPDNVEVFFCGPTGLALQLRHLCTKYGFGYRKENFPWLELAAA

</code></pre>

<p><strong>Results:</strong> </p>

<table>
<colgroup>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
</colgroup>

<thead>
<tr>
	<th style="text-align:left;">Sequences producing significant alignments:</th>
	<th style="text-align:left;">Score (Bits)</th>
	<th style="text-align:left;">E-Value</th>
</tr>
</thead>

<tbody>
<tr>
	<td style="text-align:left;">LOCUS_36720 putative oxygen-independent coproporphyrinogen III oxidase</td>
	<td style="text-align:left;">28.1</td>
	<td style="text-align:left;">0.69</td>
</tr>
</tbody>
</table>

<h3>&gt;LOCUS_36720 putative oxygen-independent coproporphyrinogen III oxidase, Length=406</h3>

<blockquote>
<p>Score = 28.1 bits (61), Expect = 0.69, Method: Compositional matrix adjust. Identities = 18/54 (33%), Positives = 30/54 (56%), Gaps = 8/54 (15%)</p>
</blockquote>

<pre><code>Query  207  LFVAMDLMHQET----KVDLITGLKSRTKTGRPDWEEIFKDVAKQHAPDNVEVF  256
            +F A+DL+HQ       +DLI+GL  +T T   DW+    + A   AP+++  +
Target  167  IFHAVDLIHQANINNWSLDLISGLPQQTLT---DWQNSL-ETAIDLAPNHISSY  216
</code></pre>

<ul>
<li><strong>Full target sequence:</strong></li>
</ul>

<pre><code>&gt;LOCUS_36720 putative oxygen-independent coproporphyrinogen III oxidase
MTLPNYFPETRSPHPTLTNDFPLPQAAYIHIPFCRRRCYYCDFAISVVGEKLRGESSQTITEYLQFLCQEIQQSP
NLGKPLETIFFGGGTPSLLDGSQLELILKTLEQKFGINSSAEISMEVDPATFDLAQLHDYQQAGVNRFSLGVQAF
QDHLLTLAGRSHCREDIFHAVDLIHQANINNWSLDLISGLPQQTLTDWQNSLETAIDLAPNHISSYDLIVEDGTP
FAKQYQPGDQPLPDDGTTAAMYVLAQEMLTNAGYTHYEISNYGQLGKQCRHNRVYWQGESYYGFGMGAASYLKGQ
RFTRPRTRSSYYQWLEDGCIIDTPVLTKDELFLETLMLGLRLKEGLNIQQLSTKFGTELATWTIENLQPYQQRGW
VELTDQQISLKDTQGFLFSNEILSTLFSRKQ
</code></pre>

<ul>
<li><strong>Target homologues:</strong></li>
</ul>

<table>
<colgroup>
<col/>
<col/>
<col/>
<col/>
<col/>
<col/>
</colgroup>

<thead>
<tr>
	<th>Protein</th>
	<th>Organism</th>
	<th>Score</th>
	<th>Query Cover</th>
	<th>E value</th>
	<th>Identity (%)</th>
</tr>
</thead>

<tbody>
<tr>
	<td>coproporphyrinogen III oxidase</td>
	<td>Coleofasciculaceae cyanobacterium SM2_1_6</td>
	<td>748</td>
	<td>99%</td>
	<td>0.0</td>
	<td>89.38</td>
</tr>
<tr>
	<td>radical SAM family heme chaperone HemW</td>
	<td>Nostoc sp. PCC 7524</td>
	<td>511</td>
	<td>95%</td>
	<td>1E-177</td>
	<td>60.62</td>
</tr>
<tr>
	<td>coproporphyrinogen III oxidase</td>
	<td>Nostocales cyanobacterium HT-58-2</td>
	<td>509</td>
	<td>94%</td>
	<td>9E-177</td>
	<td>61.30</td>
</tr>
<tr>
	<td>TPA: radical SAM family heme chaperone HemW</td>
	<td>Nostocaceae cyanobacterium</td>
	<td>504</td>
	<td>93%</td>
	<td>5E-175</td>
	<td>61.48</td>
</tr>
<tr>
	<td>radical SAM family heme chaperone HemW</td>
	<td>Brasilonema angustatum HA4187-MV1</td>
	<td>502</td>
	<td>93%</td>
	<td>3E-174</td>
	<td>61.05</td>
</tr>
<tr>
	<td>radical SAM family heme chaperone HemW</td>
	<td>Anabaena sp. UHCC 0451</td>
	<td>502</td>
	<td>94%</td>
	<td>4E-174</td>
	<td>61.52</td>
</tr>
<tr>
	<td>radical SAM family heme chaperone HemW</td>
	<td>Scytonema sp. HK-05</td>
	<td>501</td>
	<td>94%</td>
	<td>8E-174</td>
	<td>60.00</td>
</tr>
<tr>
	<td>radical SAM family heme chaperone HemW</td>
	<td>Richelia sinica</td>
	<td>501</td>
	<td>94%</td>
	<td>1E-173</td>
	<td>60.21</td>
</tr>
<tr>
	<td>radical SAM family heme chaperone HemW</td>
	<td>Brasilonema sennae</td>
	<td>501</td>
	<td>93%</td>
	<td>2E-173</td>
	<td>61.32</td>
</tr>
<tr>
	<td>radical SAM family heme chaperone HemW</td>
	<td>Mastigocladopsis repens</td>
	<td>500</td>
	<td>93%</td>
	<td>2E-173</td>
	<td>61.05</td>
</tr>
</tbody>
</table>

<p>This target seems to be the heme chaperone HemW. HemW contains three cysteines and one SAM coordinating an [4Fe–4S] cluster. In <em>Synechocystis</em>, it is involved in the heme and chlorophyll biosynthesis. Catalyzes the anaerobic oxidative decarboxylation of propionate groups of rings A and B of coproporphyrinogen III to yield the vinyl groups in protoporphyrinogen IX (<a href="https://www.uniprot.org/uniprotkb/P74132/entry">P74132</a>).</p>

<h2>Query: pNOX from <em>Chroococcidiopsis thermalis</em> | Chro_4270</h2>

<p><code>blastp -query CtNOX.fasta -db comp_0_CP12_prot_db -out results.txt</code> </p>

<ul>
<li><p><strong>Query sequence:</strong> </p>

<blockquote>
<p>CtNOX.fasta</p>
</blockquote></li>
</ul>

<pre><code>&gt; CtNOX | prokaryotik NOX (pNOX) | Chroococcidiopsis thermalis | Chro_4270 
MRSLLMKNPLTIGVSWIGLYLLLTVLPLLILLFYPPTPSDTRTFWIEFSVALGFIGLAMM
VLQFALTARINCVEASYGVDLILQFHRYTSIVAFGFILLHPVILFIDQPETLQLLNFFQA
PWRARFAVIGTLALIVLIVTSIWRKQLNIPYEPWRILHGIMAVLAVGFGLAHALGVSYYL
SLFWKAVLWGGIALVALWLLVYIRLVKPWMMLRKPYLVETVTPQRGDVWNLVLRPRGHEG
LHFQPGQFAWITLQISPFRMREHPFSFASSAEKPDRIEFGIKALGDFTKTIKDVKPGTKA
FLDGPYGVFTSDRYENTAGFIFIAGGIGITPIYSILLTLADRNDERPLTLIYSSKTREDI
TYREEIEALKDRLDLTVIHVLREPPDNWSGESGYVDKELLKRYIPKRPATRNYFICAAPK
MMDQVEQALHDLDVPVTHVHMEHYNLV
</code></pre>

<p><strong>Results:</strong> </p>

<table>
<colgroup>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
</colgroup>

<thead>
<tr>
	<th style="text-align:left;">Sequences producing significant alignments:</th>
	<th style="text-align:left;">Score (Bits)</th>
	<th style="text-align:left;">E-Value</th>
</tr>
</thead>

<tbody>
<tr>
	<td style="text-align:left;">LOCUS_14850 similar to aluminum resistance protein</td>
	<td style="text-align:left;">33.9</td>
	<td style="text-align:left;">0.019</td>
</tr>
</tbody>
</table>

<h3>&gt;LOCUS_14850 similar to aluminum resistance protein Length=411</h3>

<ul>
<li><strong>Alignment</strong></li>
</ul>

<blockquote>
<p>Score = 33.9 bits (76), Expect = 0.019, Method: Compositional matrix adjust.
 Identities = 21/67 (31%), Positives = 31/67 (46%), Gaps = 3/67 (4%)</p>
</blockquote>

<pre><code>Query  281  IKALGDFTKTIKDVKPGTKAFLDGPYGVFTSDRYENTAGFIFIAGGIGITPIYSILLT--  338
            I+ +G   + IK   P    F+D  YG F+ DR     G   +AG +   P  +I+ T  
Target  179  IEEVGKIVQLIKQQNPKVICFVDNCYGEFSEDREPTQVGVDLMAGSLIKAPGGTIITTGG  238

Query  339  -LADRND  344
             +A R D
Target  239  YVAGRAD  245

</code></pre>

<ul>
<li><strong>Full target sequence</strong></li>
</ul>

<pre><code>&gt;LOCUS_14850 similar to aluminum resistance protein
MSSMSSELVQAAVQDLQSIFSGIDNTVKSNLAKVLTAFREQRVGTQHFAGASGYGYNDSGRETLDRVFA  
QIVGAEAAIVRSQIVSGTHAITCALFGVLRPGDEMLSVMGAVHDTLEEVVGLREQGQGSLMEFGISYRE  
VPMTAEGKPDWDKIATAVSSKTKMAYIQRSCGYAWRRTLTIEEVGKIVQLIKQQNPKVICFVDNCYGEF  
SEDREPTQVGVDLMAGSLIKAPGGTIITTGGYVAGRADLVEQAACRLTAPGLGSFLGAGFELRLLFQGL  
FLAPQMAGEAIKSSHLISYVFDKLGYPVTPPPTTPRGDSIVAIKLGSPERMIAFCSAMQKQSPVGAYLE  
PIPGEMYGYNDEIIMAGGTFIQGSTSELSADGPIREPYIVYSQGGTHWTHMAIALESIIDRVGTIE  
</code></pre>

<p>It contains the Pfam domain <a href="https://www.ebi.ac.uk/interpro/entry/pfam/PF06838/">PF06838</a>: Methionine gamma-lyase. No predicted transmembrane domains. </p>

<figure>
<img src="LOCUS_14850.png" alt="" />
</figure>

<ul>
<li><strong>Target homologues:</strong></li>
</ul>

<table>
<colgroup>
<col/>
<col/>
<col/>
<col/>
<col/>
<col/>
</colgroup>

<thead>
<tr>
	<th>Description</th>
	<th>Scientific Name</th>
	<th>Total Score</th>
	<th>Query Cover</th>
	<th>E value</th>
	<th>Identity (%)</th>
</tr>
</thead>

<tbody>
<tr>
	<td>Uncharacterized protein YnbB</td>
	<td>Bacillus subtilis subsp. subtilis str. 168</td>
	<td>384</td>
	<td>95%</td>
	<td>4E-130</td>
	<td>46.58</td>
</tr>
<tr>
	<td>Uncharacterized 33.9 kDa protein in glnA 5'region</td>
	<td>Lactobacillus delbrueckii subsp. bulgaricus</td>
	<td>268</td>
	<td>72%</td>
	<td>1E-86</td>
	<td>43.96</td>
</tr>
<tr>
	<td>Uncharacterized protein in glnR 5'region</td>
	<td>Bacillus cereus</td>
	<td>82.4</td>
	<td>17%</td>
	<td>2E-18</td>
	<td>52.86</td>
</tr>
</tbody>
</table>

<p>Target is very similar to <a href="https://www.uniprot.org/uniprotkb/P94479/entry">YnbB</a> from <em>B. subtilis</em>, which is a “Cystathionine beta-lyase family protein involved in aluminum resistance. It is a pyridoxal-phosphate dependent enzyme. The homologue in <em>Listeria monocytogenes</em> is upregulated in response to chlorine dioxide (<a href="https://doi.org/10.1128/AEM.00004-1421">Pleitner et al., 2014</a>).</p>

<h2>Query: Ferric chelate reductase, G2WM87, <em>Saccharomyces cerevisiae</em></h2>

<p><code>blastp -query ferric-chelate-reductase-yeast.fasta -db comp_0_CP12_prot_db -out results.txt</code> </p>

<ul>
<li><p><strong>Query sequence:</strong> </p>

<blockquote>
<p>‌ferric-chelate-reductase-yeast.fasta</p>
</blockquote></li>
</ul>

<pre><code>&gt; tr|G2WM87|G2WM87_YEASK ferric-chelate reductase (NADPH) 
OS=Saccharomyces cerevisiae (strain Kyokai no. 7 / NBRC 101557) 
OX=721032 GN=K7_FRE7 PE=3 SV=1
MIEERDLVLSNGIHCIADIHSELYARLKKESQAATPWVYQKQYGKFVTYFVAVIIFLSLI
KKLAFMYYDSSEEFLPEKKNSPTTPSVFLARIMTKLVAFNRYICYRKFPTLIFSYLGIPT
SVGTFLVVMATTLYTLLYCFVPHPFYRPCAGFGSPPLSVRAGIMAISLVPFVFSLSRKIN
VIGWLVGLSYEKINIYHQWASILCLFFSWVHVIPFLRQARHEGGYERMHQRWKASDMWRS
GVPPILFLNLLWLSSLPIARRHFYEIFLQLHWILAVGFYISLFYHVYPELNSHMYLVATI
VVWFAQLFYRLAVKGYLRPGRSFMASTIANVSIVGEGCVELIVKDVEMAYSPGQHIFVRT
IDKGIISNHPFSIFPSAKYPGGIKMLIRAQKGFSKRLYESNDDMKKILIDGPYGGIERDI
RSFTNVYLICSGSGISTCLPFLQKYGPILHKTNLEVITLDWVVRHREDISWIRDETCTLS
NNFRQLFLDGKIVVRIYVCSDSTVPGTIKTLPQTADTASDQSDLAKREKDTEFGQDDTES
NSTFDKSNNEYKGLITIIPSKPDLNQVINDYQIGFRNCFICSGSDSLRYTVGNSVAGLQA
KVFSNKNVEECYLHSESFGY
</code></pre>

<p><strong>Results:</strong> </p>

<table>
<colgroup>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
</colgroup>

<thead>
<tr>
	<th style="text-align:left;">Sequences producing significant alignments:</th>
	<th style="text-align:left;">Score (Bits)</th>
	<th style="text-align:left;">E-Value</th>
</tr>
</thead>

<tbody>
<tr>
	<td style="text-align:left;">LOCUS_20480 hypothetical protein</td>
	<td style="text-align:left;">30.8</td>
	<td style="text-align:left;">0.36</td>
</tr>
<tr>
	<td style="text-align:left;">LOCUS_07060 hypothetical protein</td>
	<td style="text-align:left;">28.5</td>
	<td style="text-align:left;">2.1</td>
</tr>
<tr>
	<td style="text-align:left;">LOCUS_33540 hypothetical protein</td>
	<td style="text-align:left;">26.6</td>
	<td style="text-align:left;">5.4</td>
</tr>
<tr>
	<td style="text-align:left;">LOCUS_17590 ferredoxin-nitrite reductase</td>
	<td style="text-align:left;">26.2</td>
	<td style="text-align:left;">7.6</td>
</tr>
</tbody>
</table>

<h3>&gt;LOCUS_20480 hypothetical protein</h3>

<blockquote>
<p>Score = 30.8 bits (68), Expect = 0.36, Method: Compositional matrix adjust. Identities = 41/168 (24%), Positives = 72/168 (43%), Gaps = 21/168 (13%), Length=988</p>
</blockquote>

<pre><code>Query  203  LCLFFSWVHVIPFLRQARHEGGYERMHQRWKASDMWRSGVPPIL-FLNLLWLSSLPIARR  261
            LC ++   +  P + + R   G  + H+ W+++   R  V P+L F  L+ +  L +   
Target  204  LCRYYQ--NFFPAVARGRVTVGDAKYHKGWRSTQELRGYVLPLLVFFTLVQVILLLVE--  259

Query  262  HFYEIFLQLHWILAVGFYISLFYHVYPELNSHMYLVATIVVWF--AQLFYRLAV--KGYL  317
                 FL L  + ++   +S        L+  + L   +V WF  ++L  +L V     L
Target  260  -----FLYLQQVESLNLDVSSIASFLAILSGIIGLFELLVQWFISSRLIDKLGVFIAAML  314

Query  318  RPGRSFMASTIANVSIVGEGCVELIVKDVEMAYSPGQHIFVRTIDKGI  365
             PG   +AS I    ++G     +I+K ++        IF  TI  GI
</code></pre>

<ul>
<li><strong>Full target protein sequence:</strong></li>
</ul>

<pre><code>&gt;LOCUS_20480 hypothetical protein
MQYDVGRRWGAGILQSLNIRPEEVERTALMFALYATTSIGLLWLENTTLALFLTKYGTNWLPIIYVASALLSSGL
GLFYSWLEKQLPMRRLLLFISLFMALPMFLFWLGLGTTGIFLLVTVFMLRLWVDVTDILKSLNTSIAANQLFNIR
EIKRTYPLIASGVLVAELLSGFSLPLVIAAVGIKNVMFLSGVILLVGAGIMFYLCRYYQNFFPAVARGRVTVGDA
KYHKGWRSTQELRGYVLPLLVFFTLVQVILLLVEFLYLQQVESLNLDVSSIASFLAILSGIIGLFELLVQWFISS
RLIDKLGVFIAAMLLPGSLLLASVITFTGLLGSLSGVIIIKILDEIFRYTILAGIEPTLFQPLPEGMRNRYQARV
QGIVQPVAMGVTGVALWGSNLFLEQMGVQGEFYLLLLGLILASLIWLYATWQMRSTYVNLLVQSAEQGRLSTSNV
DLRAWKQAILQALEQPNNERDHRSCIQLLCQIDPTGVGDVLAPLLPKLPPRLQYQSLEAMLRYPSGYQPQVQELI
DRENNPPLDQKPNQRPLPPEVLALALRYLWLTQPDPNPNTLTPYLELSQPSTIRATAAALMLRWGTPPQKSRATD
ILRRMLTDPQEKERIMGCRALREANYLQALRIYIPRLLEDESLRVRCVLLDVIARLRLEEYYPALLRGLYYKSTR
EAARQALISMEDEAIALVRSLAADPHKPQLVRFQAWEVMGGIGTRLAVASLVEELLTSWGSTRQQILKTLLKVPG
ESGIEMVLDQLGRSGVEHLIYQELLFLAQVYGAIADLLGDELELLRQSLQDTVDDILDRCFLLMKFLYPPTAIQA
AAFNLDSEARSSIALGIEILDNTLDLPTKQVLLRVLDQRSIPERLLSLQPLLPYKKMSPRDRVHHLLELRYFLSD
WCLACCFHGARTARWGLNLDIILLCLRHPAGLVREAVLAYLQVASPRTCKSLIQLLDQDQDPLVAAQIKQLLFTK
NLEGISTQKLPNP

</code></pre>

<ul>
<li><strong>Target homologues:</strong></li>
</ul>

<table>
<colgroup>
<col/>
<col/>
<col/>
<col/>
<col/>
<col/>
</colgroup>

<thead>
<tr>
	<th>Protein</th>
	<th>Organism</th>
	<th>Total Score</th>
	<th>Coverage</th>
	<th>E-value</th>
	<th>Identity (%)</th>
</tr>
</thead>

<tbody>
<tr>
	<td>MFS transporter</td>
	<td>Coleofasciculaceae cyanobacterium SM2_1_6</td>
	<td>1139</td>
	<td>62%</td>
	<td>0.0</td>
	<td>93.03</td>
</tr>
<tr>
	<td>HEAT repeat domain-containing protein</td>
	<td>Coleofasciculus chthonoplastes</td>
	<td>1041</td>
	<td>97%</td>
	<td>0.0</td>
	<td>53.84</td>
</tr>
<tr>
	<td>HEAT repeat domain-containing protein</td>
	<td>Cyanobacteriota bacterium</td>
	<td>1038</td>
	<td>97%</td>
	<td>0.0</td>
	<td>55.48</td>
</tr>
<tr>
	<td>Npt1/Npt2 family nucleotide transporter</td>
	<td>Microcoleus sp. FACHB-831</td>
	<td>1037</td>
	<td>97%</td>
	<td>0.0</td>
	<td>55.48</td>
</tr>
<tr>
	<td>TPA: MFS transporter</td>
	<td>Cyanobacteria bacterium UBA9273</td>
	<td>1034</td>
	<td>97%</td>
	<td>0.0</td>
	<td>54.74</td>
</tr>
<tr>
	<td>ATP/ADP translocase</td>
	<td>uncultured Coleofasciculus sp.</td>
	<td>1033</td>
	<td>97%</td>
	<td>0.0</td>
	<td>56.14</td>
</tr>
<tr>
	<td>MFS transporter</td>
	<td>Kastovskya adunca ATA6-11-RM4</td>
	<td>1028</td>
	<td>97%</td>
	<td>0.0</td>
	<td>56.46</td>
</tr>
<tr>
	<td>HEAT repeat domain-containing protein</td>
	<td>Microcoleus sp. FACHB-SPT15</td>
	<td>1025</td>
	<td>97%</td>
	<td>0.0</td>
	<td>54.65</td>
</tr>
<tr>
	<td>HEAT repeat domain-containing protein</td>
	<td>Symplocastrum torsivum CPER-KK1</td>
	<td>1023</td>
	<td>97%</td>
	<td>0.0</td>
	<td>55.26</td>
</tr>
<tr>
	<td>TPA: MFS transporter</td>
	<td>Cyanobacteria bacterium UBA11370</td>
	<td>1015</td>
	<td>97%</td>
	<td>0.0</td>
	<td>53.25</td>
</tr>
</tbody>
</table>

<p>Most similar homologue is an MFS transporter from <em>Coleofasciculaceae cyanobacterium</em> SM2_1_6. No functional data is available. The target contains an Armadillo-like fold. Armadillo (Arm) repeat proteins contain tandem copies of a degenerate protein sequence motif that forms a conserved three-dimensional structure. Animal Arm repeat proteins function in various processes, including intracellular signalling and cytoskeletal regulation.</p>

<figure>
<img src="LOCUS_20480.png" alt="" />
</figure>

<h2>Query: pNOX from <em>Streptomyces avermitilis</em></h2>

<p><code>blastp -query Q82R51Oxidoreductase.fasta -db comp_0_CP12_prot_db -out results.txt</code> </p>

<ul>
<li><p><strong>Query sequence:</strong> </p>

<blockquote>
<p>Q82R51Oxidoreductase.fasta</p>
</blockquote></li>
</ul>

<pre><code>&gt; tr|Q82R51|Q82R51_STRAW Oxidoreductase OS=Streptomyces 
avermitilis (strain ATCC 31267 / DSM 46492 / JCM 5070 / NBRC 14893 /
NCIMB 12804 / NRRL 8165 / MA-4680) OX=227882 GN=SAVERM_293 PE=4 
SV=1, Length=449
MPVLIVIAIYAVRIARAPLDPRLKKRHVGPLSVTTGAVMLYLAWDWLRPEGLSAVYFVGE
FAGVLSAYLMSCTLVLATRLMWLEQWFGGLDRMYRQHKRYAVWSILLLTPHLLLHFFSGF
DGSQYGYAHRTASVGMGHLLGAVSAIGLLLLVLISLGQVGRILRLPYERWLFLHRLTGLL
LLSALLHGWFLDLIINGSTPLLAIYVTMATVGMTAYAYDELVLRHREPRADYTIHRVERP
TPDILDLTLTPTGKTALPVTGGQFVYLRVGGWHEHPFSVAGTQADGSVRLTIRALGRGTR
GLYTDVSEGHPATLKGPYGMFDHTLGGPRQIWIAGGIGIAPFLGWLTHPGAEPAQTDLFY
CAATAEDAPFLSELTAAAAHRPEFRLHPTFSRSHGRLTAERIQAEAGPITPDTHVFLCGP
ASMIENLTRGLHRQGVPRQHLHAEHFAFR
</code></pre>

<p><strong>Results:</strong> </p>

<table>
<colgroup>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
</colgroup>

<thead>
<tr>
	<th style="text-align:left;">Sequences producing significant alignments:</th>
	<th style="text-align:left;">Score (Bits)</th>
	<th style="text-align:left;">E-Value</th>
</tr>
</thead>

<tbody>
<tr>
	<td style="text-align:left;">LOCUS_05450 oxidoreductase FAD/NAD(P)-binding domain protein</td>
	<td style="text-align:left;">32.7</td>
	<td style="text-align:left;">0.047</td>
</tr>
</tbody>
</table>

<h3>LOCUS_05450 oxidoreductase FAD/NAD(P)-binding domain protein, Length=405</h3>

<blockquote>
<p>Score = 32.7 bits (73), Expect = 0.047, Method: Compositional matrix adjust. Identities = 48/192 (25%), Positives = 73/192 (38%), Gaps = 37/192 (19%)</p>
</blockquote>

<pre><code>
Query  277  FSVAGTQ-----ADGSVRLTIRAL-------GRGTRGLYT----DVSEGHPATLKGPYGM  320
            +S+A T+      D +V L +R L       G   +G+ +    ++  G    + GP G 
Target  185  YSIASTRHGDRGDDKTVSLCVRQLEYKDPATGEKVQGVCSTFLCNLQPGDDLKITGPVGK  244

Query  321  FDHTLGGPRQ--IWIAGGIGIAPFLGWL------THPGAE-PAQTDLFYCAATAEDAPFL  371
                   P    I +A G GIAPF  +L      TH   +   +  LF+  A   +  + 
Target  245  EMLLPEDPESNVIMMATGTGIAPFRAYLWRMFKETHEDYKFKGKAWLFFGVAYTPNILYK  304

Query  372  SELTAAAAHRPE-FRLHPTFSRSHGRLTA------ERIQAEAGPI-----TPDTHVFLCG  419
             EL A  A  P+ FR+    SR              RI   A  +        TH ++CG
Target  305  EELEALQAEFPDQFRITYAISREQKNADGSKVYIQSRIAEHADEVWEWIQDEKTHTYICG  364

Query  420  PASMIENLTRGL  431
               M   + +GL
Target  365  LKGMEGGIDQGL  376
</code></pre>

<h3>Full length protein</h3>

<pre><code>    &gt;LOCUS_05450 oxidoreductase FAD/NAD(P)-binding domain protein
    MNILSTAVSANNTESGGRMFVYEVVGLRQNQASDRANYSIRKSGSVFITVAYNRMNEEMRR
    IARLGGKILSIKPLVSPAEFIPQGSVLATETAPASSPEVPQEAPAKAPKKEKRDVPVNTYR
    PNKPLMGKCLSNEELVREGGEGTVRHLTFDLEGSDLHYLEGQSIGIIPDGEDKNGKPHKLR
    LYSIASTRHGDRGDDKTVSLCVRQLEYKDPATGEKVQGVCSTFLCNLQPGDDLKITGPVGK
    EMLLPEDPESNVIMMATGTGIAPFRAYLWRMFKETHEDYKFKGKAWLFFGVAYTPNILYKE
    ELEALQAEFPDQFRITYAISREQKNADGSKVYIQSRIAEHADEVWEWIQDEKTHTYICGLK
    GMEGGIDQGLSGAAGKFGITWSDYQKQLKKADRWHVETY
</code></pre>

<ul>
<li><strong>InterPro - Classification of protein families</strong></li>
</ul>

<figure>
<img src="LOCUS_05450.png" alt="" />
</figure>

<ul>
<li><strong>Target homologues:</strong></li>
</ul>

<table>
<colgroup>
<col/>
<col/>
<col/>
<col/>
<col/>
<col/>
</colgroup>

<thead>
<tr>
	<th>Protein</th>
	<th>Organism</th>
	<th>Score</th>
	<th>Coverage</th>
	<th>E-value</th>
	<th>Identity (%)</th>
</tr>
</thead>

<tbody>
<tr>
	<td>Ferredoxin--NADP reductase</td>
	<td>Thermosynechococcus vestitus BP-1</td>
	<td>533</td>
	<td>97%</td>
	<td>0.0</td>
	<td>64.48</td>
</tr>
<tr>
	<td>Ferredoxin--NADP reductase</td>
	<td>Trichormus variabilis ATCC 29413</td>
	<td>528</td>
	<td>97%</td>
	<td>0.0</td>
	<td>61.89</td>
</tr>
<tr>
	<td>Ferredoxin--NADP reductase</td>
	<td>Nostoc sp. PCC 7120 = FACHB-418</td>
	<td>525</td>
	<td>97%</td>
	<td>0.0</td>
	<td>61.66</td>
</tr>
<tr>
	<td>Ferredoxin--NADP reductase</td>
	<td>Nostoc sp. PCC 7119</td>
	<td>525</td>
	<td>97%</td>
	<td>0.0</td>
	<td>61.11</td>
</tr>
<tr>
	<td>Ferredoxin--NADP reductase</td>
	<td>Synechocystis sp. PCC 6803 substr. Kazusa</td>
	<td>520</td>
	<td>98%</td>
	<td>0.0</td>
	<td>61.37</td>
</tr>
<tr>
	<td>Ferredoxin--NADP reductase</td>
	<td>Picosynechococcus sp. PCC 7002</td>
	<td>511</td>
	<td>99%</td>
	<td>1E-180</td>
	<td>61.67</td>
</tr>
<tr>
	<td>Ferredoxin--NADP reductase</td>
	<td>Spirulina sp.</td>
	<td>460</td>
	<td>72%</td>
	<td>2E-162</td>
	<td>75.43</td>
</tr>
<tr>
	<td>Ferredoxin--NADP reductase</td>
	<td>Cyanophora paradoxa</td>
	<td>379</td>
	<td>78%</td>
	<td>3E-129</td>
	<td>57.86</td>
</tr>
<tr>
	<td>Ferredoxin--NADP reductase</td>
	<td>Oryza sativa Indica Group</td>
	<td>369</td>
	<td>73%</td>
	<td>1E-125</td>
	<td>57.19</td>
</tr>
<tr>
	<td>Ferredoxin--NADP reductase</td>
	<td>Oryza sativa Japonica Group</td>
	<td>369</td>
	<td>73%</td>
	<td>3E-125</td>
	<td>57.19</td>
</tr>
</tbody>
</table>

<p>This target is most likely the ferredoxin-NADP reductase (FNR), which is homologous to NOXs.</p>

<h2>Query: Superoxide-generating NADPH oxidase from <em>Dictyostelium discoideum</em></h2>

<p><code>‌blastp -query Q54F44_NOXC_Dycto.fasta -db comp_0_CP12_prot_db -out results_NOXC_blastp.txt</code></p>

<ul>
<li><strong>Query sequence:</strong>
 &gt; Q54F44_NOXC_Dycto.fasta</li>
</ul>

<pre><code>&gt;sp|Q54F44|NOXC_DICDI Superoxide-generating NADPH oxidase heavy 
chain subunit C | OS=Dictyostelium discoideum OX=44689 GN=noxC PE=2 
SV=1    
MEKQKNISIKKFANPLSFPHEFRTNLHKSTFLDASLLELPNISVDKTKNENNFSITDKILMESLKENE
FKIREFIKNQESGNRVLMIDKNSLKELFNINDNQALDLIFNQYVQSTRPNLIRKSSNSSLNIDLKKEI
KKKTKKSRKSFSRNSKLNNNDSKIDKNDNYIDKINDINNFNSDIEEIFKKKKEIENTLKIPIGGNSFP
IPVEFNNNNNNNNNNNNNNNNNYNNNIDNNNNNNNNNNNNNINNCNNNNINNDNNNNNNNDNDNNNIN
TVDNHDDDIINNSNNFNKNEYPSSNISPISPKSSISSFPTNLNNSINNTGSMVSDSLSSCRNSISSSS
IDSSVASIPITIQSIDFEDKNIKSDQFKIISKSNIENTIETNPIPPFNQTNNQCEVQLQSHSLPTILK
QPHIYKSKSFSSSINSNSKIKKIKKSRSFEIESKINLFDVINHIYLNLSKVGSEEQKITSVFKLYDIY
DKGFISRDDLKEVLNYRTKQNGLKFQDFTMESLIDHIFQQFDKNMDGYIDFEEFKSELTINNENKVKE
KEENTNYNFKEENIGIYTEKESFHSLKRYLKIEGSKLFFISLFFIINSILVITSFLNVHANNKRAIEL
FGPGVYITRIAAQLIEFNAAIILMTMCKQLFTMIRNTKFKFLFPVDKYMTFHKLIGYTLIIASFLHTI
GWIVGMAVATGKPDNIFYDCLAPHFKFRPTVWEMIFNSLPGVTGFIMISFLIIMAILSLKIIRKSNFE
LFYYSHHLFIGFYVLLILHGTMGWIRPPTFWKWFIVPGFFYTVDRSFRLFKRTHRVEVLDYCLKNERV
INLTFSKPPSFDYKPGQYLLINVPHISKLQWHPFTMTSSPLEDKIYVHIRVTGNWTKKLFRWLSIKKQ
LQQQQQLYNNIKQQNVLPDGSNFIINNNNNIDQIDLEIGLKPFRINIDGPFGSSSQYALKQKQVILVG
AGIGVSPMASLLKDISLKKQRLQLLNQGDQIALEQSKNEITTKFGLGNLEKVHFFWLNRDQHSFQWFE
DLLIDISTNGNSNLPKISINTFNTRVFPKNDVRVFMLWNGLDKLFKAQGLDPTTNLPFKTHWGRPNWD
TIFQYYSKKYSGESISVFCCGPSQLSKELYEKCRYYTCLKTGGTKFYFHKEF
</code></pre>

<ul>
<li><strong>Results:</strong></li>
</ul>

<table>
<colgroup>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
<col style="text-align:left;"/>
</colgroup>

<thead>
<tr>
	<th style="text-align:left;">Sequences producing significant alignments:</th>
	<th style="text-align:left;">Score (Bits)</th>
	<th style="text-align:left;">E-Value</th>
</tr>
</thead>

<tbody>
<tr>
	<td style="text-align:left;">LOCUS_09080 hypothetical protein</td>
	<td style="text-align:left;">30.4</td>
	<td style="text-align:left;">0.70</td>
</tr>
<tr>
	<td style="text-align:left;">LOCUS_26610 twin-arginine translocation protein</td>
	<td style="text-align:left;">29.3</td>
	<td style="text-align:left;">0.71</td>
</tr>
<tr>
	<td style="text-align:left;">LOCUS_01810 transaldolase/EF-hand domain-containing protein</td>
	<td style="text-align:left;">30.0</td>
	<td style="text-align:left;">0.96</td>
</tr>
</tbody>
</table>

<h3>LOCUS_09080 hypothetical protein, Length=403</h3>

<blockquote>
<p>Score = 30.4 bits (67), Expect = 0.70, Method: Compositional matrix adjust. Identities = 16/72 (22%), Positives = 36/72 (50%), Gaps = 8/72 (11%)</p>
</blockquote>

<pre><code>Query  460  VGSEEQKITSVFKLYDIYDKGFISRDDLKEVLNYRTKQNGLKFQDFTMESLIDHIFQQFD  519
            V  +++++   FKL+D    G+IS ++L+  +                ++ ++ + +  D
Target  328  VDDQDERLVERFKLFDADGSGYISLEELQVCIR--------DLDALVTKAEVEAMLKGID  379

Query  520  KNMDGYIDFEEF  531
             + DG I++EEF
Target  380  VDGDGQINYEEF  391
</code></pre>

<ul>
<li><strong>Full target sequence:</strong> The same as for human DUOX.</li>
</ul>

<pre><code>&gt;LOCUS_09080 hypothetical protein
MLSNTNGSVQQELIRYMFRRDANGQYDAMACLQLWFGKSHDTDNEIRHRFGALVEQALAGELDHWMETPHGCVAL
MILLDQFPRNLYRHTVQMYSGDQKAMQIVAQKHDWQAELTPEEYLFVPCLILTHQENLEAQKQCVNYYEQIEPIL
QPTFHIFRTIFEEHLKIINLCGVFPHRDHYYGRETSATGKMLMEDSTLRFDLPLVAENGSVRFGTDSTRLWKATA
HAFDVLDRLDSLATQHQANDGKPVTIDWMTDEQIAFCKDTFRKFDKDGSGHLSAAELRSVLQSVGRPYSAERIQA
AMDLITGSENSSSISFEEFATLLQSNLVDDQDERLVERFKLFDADGSGYISLEELQVCIRDLDALVTKAEVEAML
KGIDVDGDGQINYEEFYHLFQNLKAQRA
</code></pre>

<ul>
<li><strong>Target homologues:</strong> The same Calmodulin as for human DUOX.</li>
</ul>
</body>
</html>
